### ISSUE TITLE
etl::vector with etl::string not working

### ISSUE THREAD

**Issue by fabssor** [ETLCPP/etl#391]
Hello, 

I have the following problem.

This code sample works:

```
etl::vector<int, 2> test;

test.repair();
```
If I use an `etl::string` as datatype in a vector things break. For example this:

```
etl::vector<etl::string<50>, 2> test;

test.repair();
```
Fails with: terminate called after throwing an instance of 'etl::vector_incompatible_type'

Is it not possible to use etl::string as datatype of etl::vector?

Thanks!

**Comment by jwellbelove** [comment]
repair() is a low level function that is only valid for trivially constructable types, which etl::string is not, which is why you are getting the error.

**Comment by fabssor** [comment]
So is it illegal to memcpy then a etl::vector with a etl::string?

**Comment by jwellbelove** [comment]
A non-trivially constructible object cannot be copied using `memcpy`, as it may result in _undefined behaviour_, due to the object's copy constructor not being invoked.

Using functions such as memcpy, memmove, memclr etc on non-trivially constructible objects can cause all sorts of weird bugs to occur. 
I had to track one down once, when a colleague had decided that the best way to clear an object was `memclr`. It worked fine until a `std::string` was added as a member...
